teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +195,389 @@ In addition to grouping related resources, Tekton Results provides long term sto
 of `PipelineRuns`, `TaskRuns` and `Runs` away from the runtime storage in etcd.
 Read more in [TEP-0021][tep-0021].
 
-## Open Questions
+## Proposal
 
-* [Qn #1][qn-1]: What fields will no longer be available? What do we lose from the optimizations?
-  We will discuss this in the proposal.
-* [Qn #2][qn-1]: What does an example status look like?
-  We will provide examples in the proposal.
+The `PipelineRunStatus` should contain the names and overall status (`ConditionSucceeded`)
+of its `TaskRuns` and `Runs`. Users and tools can find the complete status of `TaskRuns`
+and `Runs` in the cluster. Moreover, they can rely on [Tekton Results](#tekton-results-api)
+and [Owner References and Labels](#owner-references-and-labels) to identify related objects.
+
+### API Changes
+
+#### 1. Add Minimal Embedded Status
+
+Introduce two new structs to store the minimal status of `TaskRuns` and `Runs` in the
+`PipelineRun` status, with the names and overall status only:
+
+```go
+type PipelineRunTaskRunMinimalStatus struct {
+	TaskRunName string `json:"taskRunName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+
+type PipelineRunRunMinimalStatus struct {
+	RunName string `json:"runName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+```
+
+Add the new fields to [`PipelineRunStatusFields`](#pipelinerun-status) as follows:
+
+```go
+type PipelineRunStatusFields struct {
+	...
+	TaskRuns map[string]*PipelineRunTaskRunStatus `json:"taskRuns,omitempty"`
+	TaskRunsStatuses []PipelineRunTaskRunMinimalStatus `json:"taskRunsStatuses,omitempty"`
+	Runs map[string]*PipelineRunRunStatus `json:"runs,omitempty"`
+	RunsStatuses []PipelineRunRunMinimalStatus `json:"runsStatuses,omitempty"`
+	...
+}
+```
+
+The existing fields providing the complete `TaskRun` and `Runs` are maps with the
+resource names as keys. However, the new fields are sub-objects instead of maps as
+recommended by the [Kubernetes API conventions][subobjects].
+
+Note that the names of `TaskRuns` and `Runs` are concatenations of the names of the
+`PipelineRuns` and `PipelineTasks`. Therefore, we have excluded the `PipelineTasks`
+names in the minimal embedded statuses. We can add this field later if needed.
```

Comment:
updated the proposal - we now support the `PipelineTasks` names too in the minimal statuses because @AlanGreene pointed out that the `(Task)Runs` names are truncated when they are too long